CONTRIBUTING.md

```diff
@@ -0,0 +1,101 @@
+
+## How to submit a change
+
+If you see something worth adding, make the relevant changes in a fork of the source code and [submit a pull request to the project](fs2-pulls). If you don't know what you could help with, take a look at [the issues marked as "help wanted"][low-hanging-fruit] or ask on [Gitter][fs2-dev].
+
+We follow similar rules to [the cats-effect project's](https://github.com/typelevel/cats-effect#development).
+Most importantly, any contributions are expected to be made in the form of [GitHub pull requests to the FS2 repository][fs2-pulls].
+Usually it takes two approvals for a change to be merged. If it takes too long to get it approved, feel free to ask on [the Gitter channel][fs2-dev].
+
+Remember to follow the [code of conduct][coc] in online and offline discourse.
+
+## Building the project locally
+
+### Prerequisites
+
+You'll need JDK 11 and [sbt][sbt].
+
+We use several sbt plugins to build and check the project, including [MiMa (Migration Manager)][mima], [scalafmt][scalafmt] and [sbt-microsites][sbt-microsites].
+
+### Build process
+
+To compile the code for the whole repository, you can start an interactive sbt shell:
+
+```bash
+$ sbt
+[info] Loading global plugins from /Users/contributor/.sbt/1.0/plugins
+[info] Loading settings for project fs2-build from plugins.sbt,metals.sbt ...
+[info] Loading project definition from /Users/contributor/dev/fs2/project
+[info] Loading settings for project root from version.sbt,build.sbt ...
+[info] Set current project to root (in build file:/Users/contributor/dev/fs2/)
+sbt:root>
+```
+
+Inside the shell, you can compile the sources for the currently selected Scala version using the `compile` command.
+To compile the code for all Scala versions enabled in the build, use `+compile`. To include tests, `test:compile` or `+test:compile`, accordingly.
+
+
+### Testing
+
+To test the code, you can run the `testJVM` or `testJS` command, depending on the platform you want to run the tests on.
+
+Before submitting a change for review, it's worth running some extra checks that will be triggered in Continuous Integration:
+
+```sbt
+sbt:root> fmtCheck; testJVM; testJS; doc; mimaReportBinaryIssues; docs/mdoc; microsite/makeMicrosite
+```
+
+That will check the formatting, run all tests on the JVM and JS platforms, report any binary compatibility issues (as detected by [MiMa][mima]) and build the site.
+
+If you run into any problems with tests, binary compatibility or other issues, feel free to ask questions on [the Gitter channel][fs2-dev].
+
+### Website
+
+To build the microsite, use the `microsite/makeMicrosite` task in sbt. Then, go to `site/target/site`, and run a HTTP server on the port of your choice:
+
+```bash
+sbt:root> microsite/makeMicrosite
+info: Compiled in 24s (0 errors, 0 warnings)
+[info] Configuration file: /Users/contributor/dev/fs2/site/target/scala-2.13/resource_managed/main/jekyll/_config.yml
```

Comment:
are any other developer requirements necessary (ruby / jekyll / etc)?

Curious as I have everything on my laptop but not sure what else is needed for microsite

Reply:
Yeah, pretty sure we'll also need jekyll to build the microsite. I'll mention that.

Reply:
I'll try building it in docker on a clean image and see what else is missing :)

Reply:
okay, basically we were missing jekyll and node (for running scalajs).